src/test/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImplTest.java

```diff
@@ -0,0 +1,418 @@
+    }
+
+
+    @Test
+    void testUpdateCaseWithMultipleHearingsWithCourtroomInRequest() {
+        CourthouseEntity courthouseEntity = CommonTestDataUtil.createCourthouse(SWANSEA);
+        CourtroomEntity courtroomEntity = CommonTestDataUtil.createCourtroom(courthouseEntity, "1");
+        CaseEntity existingCaseEntity = CommonTestDataUtil.createCase("case1", courthouseEntity);
+        existingCaseEntity.setId(1);
+
+        Mockito.when(caseRepository.findByCaseNumberAndCourthouse_CourthouseName(anyString(), anyString()))
+            .thenReturn(Optional.of(existingCaseEntity));
+        Mockito.when(courthouseRepository.findByCourthouseName(any())).thenReturn(Optional.of(courthouseEntity));
+        Mockito.when(courtroomRepository.findByNames(any(), any())).thenReturn(courtroomEntity);
+
+        Mockito.when(caseRepository.saveAndFlush(any())).thenAnswer(invocation -> {
+            Object[] args = invocation.getArguments();
+            return args[0];
+        });
+
+        List<HearingEntity> existingHearings = Lists.newArrayList(CommonTestDataUtil.createHearing(
+            existingCaseEntity,
+            courtroomEntity,
+            LocalDate.now()
+        ), CommonTestDataUtil.createHearing(
+            existingCaseEntity,
+            CommonTestDataUtil.createCourtroom(courthouseEntity, "2"),
+            LocalDate.now()
+        ));
+
+        Mockito.when(hearingRepository.findAll()).thenReturn(existingHearings);
+        Mockito.when(hearingRepository.saveAndFlush(any())).thenAnswer(invocation -> {
+            Object[] args = invocation.getArguments();
+            return args[0];
+        });
+
+        AddCaseRequest request = CommonTestDataUtil.createUpdateCaseRequest("1");
+        service.addCaseOrUpdate(request);
+
+        Mockito.verify(caseRepository).saveAndFlush(caseEntityArgumentCaptor.capture());
+        Mockito.verify(hearingRepository).saveAndFlush(hearingEntityCaptor.capture());
+
+        CaseEntity updatedCaseEntity = caseEntityArgumentCaptor.getValue();
+        HearingEntity updatedHearingEntity = hearingEntityCaptor.getValue();
+
+        assertEquals(SWANSEA, updatedCaseEntity.getCourthouse().getCourthouseName());
+        assertEquals("case1", updatedCaseEntity.getCaseNumber());
+        assertEquals(3, updatedCaseEntity.getDefendants().size());
+        assertEquals(3, updatedCaseEntity.getProsecutors().size());
+        assertEquals(3, updatedCaseEntity.getDefenders().size());
+
+
+        assertEquals("1", updatedHearingEntity.getCourtroom().getName());
+        assertEquals(SWANSEA, updatedHearingEntity.getCourtroom().getCourthouse().getCourthouseName());
+        assertEquals(1, updatedHearingEntity.getJudges().size());
+        assertEquals(LocalDate.now(), updatedHearingEntity.getHearingDate());
+        assertNotNull(updatedHearingEntity.getCourtCase());
+    }
+
+    @Test
+    void testUpdateCaseWithMultipleHearingsWithoutCourtroomInRequest() {
```

Comment:
Is there a test which does update case with multiple (or just one) hearings with a matching date, but not a matching courtroom?

It should create a new hearing for the same date with a different courtroom.

Reply:
I don't think we have test that covers that exact scenario, we have the test testUpdateCaseWithMultipleHearingsWithCourtroomInRequest which has two courtrooms(1,2) and a hearing for each. We then do an update request for courtroom 2 and check if the case is updated and related hearing is updated.

Reply:
Added a new test testUpdateCaseWithMultipleHearingsWithOldHearingDateWithCourtroomInRequest for the requested scenario.

Reply:
I'm not sure that test covers the scenario above? It looks as if hearings exist for courtroom 1 and 2, then the request is for courtroom 1 again?

We need a hearing match on date only and check that it creates a new hearing with the specified courtroom.

Reply:
I covered the inverse of what you suggested previously, so I have created a new Test testUpdateCaseWithNonExistingCourtroomAndMatchingHearingDate. This will create a courtroom with an existing case and hearing for todays date. Then an update request will be made with a different , non existing, courtroom. This should match on hearing_date but not on courtroom thus creating a new hearing.